tests/integration/buildcmd/test_build_terraform_applications.py

```diff
@@ -50,20 +60,83 @@ def setUpClass(cls):
             ):
                 LOG.info(log)
 
-    def setUp(self):
-        super().setUp()
-        shutil.rmtree(Path(self.working_dir))
-        shutil.copytree(Path(self.terraform_application_path), Path(self.working_dir))
-
     def run_command(self, command_list, env=None, input=None, timeout=None):
-        process = Popen(command_list, stdout=PIPE, stderr=PIPE, stdin=PIPE, env=env, cwd=self.working_dir)
+        process = Popen(
+            command_list, stdout=PIPE, stderr=PIPE, stdin=PIPE, env=env, cwd=self.terraform_application_execution_path
+        )
         try:
             (stdout, stderr) = process.communicate(input=input, timeout=timeout)
             return stdout, stderr, process.returncode
         except TimeoutExpired:
             process.kill()
             raise
 
+    @pytest.fixture(scope="class", autouse=True)
+    def build_with_prepare_hook(self):
+        if not self.function_identifier:
+            # This doesn't make sense for all tests that inherit this base class, skip for those
+            return
+
+        command_list_parameters = {"function_identifier": self.function_identifier, "hook_name": "terraform"}
+        if self.build_in_container:
+            command_list_parameters["use_container"] = True
+            command_list_parameters["build_image"] = self.docker_tag
+            if self.override:
+                command_list_parameters[
+                    "container_env_var"
+                ] = "TF_VAR_HELLO_FUNCTION_SRC_CODE=./artifacts/HelloWorldFunction2"
```

Comment:
Will `/` be a problem on Windows?

Reply:
This shouldn't be an issue. This part of the code was just moved and isn't anything new.